src/js_native_api.h

```diff
@@ -402,12 +402,14 @@ NAPI_EXTERN napi_status NAPI_CDECL napi_create_arraybuffer(napi_env env,
                                                            void** data,
                                                            napi_value* result);
 NAPI_EXTERN napi_status NAPI_CDECL
+#ifndef NODE_API_NO_EXTERNAL_BUFFERS_ALLOWED
```

Comment:
This line should be moved one line up, above the `NAPI_EXTERN napi_status NAPI_CDECL`, as it is part of the declaration.

Maybe we should simply add a testing addon that defines `NODE_API_NO_EXTERNAL_BUFFERS_ALLOWED` and it can correctly compile?

Reply:
@legendecas thats a good idea, I had manually tested defining it, but I only checked that I got an error, not that would not get an error.  I'll plan to add that.

Reply:
@legendecas fixed and added to the two flavors of test_general